---
title: "R Notebook"
output:
  html_document:
    df_print: paged
---

```{r, setup}
knitr::opts_chunk$set(engine.opts = "-i")
```


```{r}
library(tidyverse)
```

# Data prepatation

## dowloading genome anootation from EnesemleFungi
```{bash}
wget https://ftp.ebi.ac.uk/ensemblgenomes/pub/current/fungi/species_EnsemblFungi.txt
```
```{r}
metadata <- read_tsv("species_EnsemblFungi.txt")
metadata %>% distinct(genebuild)

metadata %>% 
  filter(grepl("DOE", genebuild)) %>%
  filter(grepl("fungi_ascomycota5_collection", core_db)) -> filtered

#link examples
#https://ftp.ebi.ac.uk/ensemblgenomes/pub/release-56/fungi/gff3/fungi_ascomycota3_collection/aspergillus_ochraceoroseus_ibt_24754_gca_002846915//Aspergillus_ochraceoroseus_ibt_24754_gca_002846915.Aspergillus_ochraceoroseus_IBT_24754_v1.1.56.gff3.gz
#https://ftp.ebi.ac.uk/ensemblgenomes/pub/release-56/fungi/fasta/fungi_ascomycota3_collection/aspergillus_ochraceoroseus_ibt_24754_gca_002846915/dna/Aspergillus_ochraceoroseus_ibt_24754_gca_002846915.Aspergillus_ochraceoroseus_IBT_24754_v1.1.dna.toplevel.fa.gz

link = "https://ftp.ebi.ac.uk/ensemblgenomes/pub/release-56/fungi"
collection = "fungi_ascomycota5_collection" 
format = "gff3"

filtered %>%
  mutate(assemly_mut = str_replace_all(string = assembly, pattern = " ", replacement = "_")) %>%
  mutate(species_mut = str_to_sentence(species)) %>%
  mutate(url_gff = paste(link, "gff3", collection, species, paste(species_mut, assemly_mut, "56.gff3.gz", sep = "."), sep = "/")) %>% 
  mutate(url_fasta = paste(link, "fasta", collection, species, "dna", paste(species_mut, assemly_mut, "dna.toplevel.fa.gz", sep = "."), sep = "/")) %>% 
  select(url_gff, url_fasta) -> urls


urls %>% select(url_gff) %>% write_delim(file = "url_gff", col_names = FALSE)
urls %>% select(url_fasta) %>% write_delim(file = "url_fasta", col_names = FALSE)

```

```{bash}
#if [ -d gff ]; then rm -Rf gff; fi #uncomment if you want to delete exisigin files
#if [ -d fasta ]; then rm -Rf fasta; fi #uncomment if you want to delete exisigin files

mkdir -p fasta_dna
mkdir -p gff
ls -la
```
```{bash, include=FALSE}
wget --no-check-certificate -P gff -i url_gff
```

```{bash, message='hide'}
wget --no-check-certificate -P fasta_dna -i url_fasta 
```

```{bash}
gunzip gff/*
gunzip fasta_dna/*
```



## Example of using AGAT tool to extract genomic features
documentation : https://agat.readthedocs.io/en/latest/tools/agat_sp_extract_sequences.html#
install using: conda install -c bioconda agat

![A](https://agat.readthedocs.io/en/latest/_images/agat_sp_extract_sequences_2.png "AGAT tool"){width=200%, height=150%}
![A](https://agat.readthedocs.io/en/latest/_images/agat_sp_extract_sequences_1.png "AGAT tool"){width=300%} 
```{bash}

conda activate genomics #activate conda genomics environment with agat tool

```
 
```{bash}
my_features="CDS_2"
mkdir -p $my_features #make directory where you want to save your features
conda activate genomics #activate conda genomics environment with agat tools

for FILE in gff/*; do dna_file=`basename $FILE| sed "s/.56.gff3$/.dna.toplevel.fa/"`; agat_sp_extract_sequences.pl -g $FILE -f fasta_dna/$dna_file -t cds -o $my_features/${dna_file}.${my_features};  done;
```

```{bash}
#5'UTR
my_features="5_UTR"
mkdir -p $my_features
conda activate genomics
for FILE in gff/*; do dna_file=`basename $FILE| sed "s/.56.gff3$/.dna.toplevel.fa/"`; agat_sp_extract_sequences.pl -g $FILE -f fasta_dna/$dna_file -t "five_prime_UTR" -o $my_features/${dna_file}.${my_features};  done;
```

```{bash}
#3'UTR
my_features="3_UTR"
mkdir -p $my_features
conda activate genomics
for FILE in gff/*; do dna_file=`basename $FILE| sed "s/.56.gff3$/.dna.toplevel.fa/"`; agat_sp_extract_sequences.pl -g $FILE -f fasta_dna/$dna_file -t "three_prime_UTR" -o $my_features/${dna_file}.${my_features};  done;
```

```{bash}
#Promoter
my_features="Promoter"
mkdir -p $my_features
conda activate genomics
for FILE in gff/*; do dna_file=`basename $FILE| sed "s/.56.gff3$/.dna.toplevel.fa/"`; agat_sp_extract_sequences.pl -g $FILE -f fasta_dna/$dna_file -t cds -eo --up 1000 -o $my_features/${dna_file}.${my_features};  done;
```

```{bash}
#Terminator
my_features="Terminator"
mkdir -p $my_features
conda activate genomics
for FILE in gff/*; do dna_file=`basename $FILE| sed "s/.56.gff3$/.dna.toplevel.fa/"`; agat_sp_extract_sequences.pl -g $FILE -f fasta_dna/$dna_file -t CDS -eo --down 500 -o $my_features/${dna_file}.${my_features};  done;
```
